buildSrc/src/main/kotlin/otel.javaagent-instrumentation.gradle.kts

```diff
@@ -1,142 +1,23 @@
-import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
-import io.opentelemetry.instrumentation.gradle.bytebuddy.ByteBuddyPluginConfigurator
-
 plugins {
-  id("net.bytebuddy.byte-buddy")
-
-  id("otel.java-conventions")
-  id("otel.shadow-conventions")
+  id("otel.javaagent-testing")
+  id("otel.publish-conventions")
 }
 
-extra["mavenGroupId"] = "io.opentelemetry.javaagent.instrumentation"
-// Shadow is only for testing, not publishing.
-extra["noShadowPublish"] = true
+// TODO(anuraaga): Migrate MuzzlePlugin to kotlin script since Java plugins don't seem to be usable" +
+// with plugins block. It will be good to add a property for bootstrapRuntime to that plugin at the
+// time instead of defining it here and reading it in a non-obvious way.
+apply(plugin = "muzzle")
 
-val skipPublish: Boolean? by extra
-
-if (skipPublish != true) {
-  apply(plugin = "otel.publish-conventions")
-}
-
-apply(plugin = "otel.instrumentation-conventions")
-
-if (projectDir.name == "javaagent") {
-  apply(plugin = "muzzle")
-
-  base.archivesBaseName = projectDir.parentFile.name
-}
+extra["mavenGroupId"] = "io.opentelemetry.javaagent.instrumentation"
 
-val toolingRuntime by configurations.creating {
-  isCanBeConsumed = false
-  isCanBeResolved = true
-}
+base.archivesName.set(projectDir.parentFile.name)
 
+// Used by muzzle
 val bootstrapRuntime by configurations.creating {
   isCanBeConsumed = false
   isCanBeResolved = true
 }
 
-afterEvaluate {
-  // TODO(anuraaga): There is no obvious reason this needs to be in afterEvaluate but it breaks
```

Comment:
Was a fun one - `afterEvaluate` allowed testing-common -> armeria-shaded to be at the end of the classpath, so the accidentally bundled in Jackson was winning in priority